pkgs/by-name/sw/switch-to-configuration-ng/src/src/main.rs

```diff
@@ -219,29 +219,36 @@ fn get_active_units(
                 id,
                 _description,
                 _load_state,
-                active_state,
-                sub_state,
+                state,
+                substate,
                 following,
-                _unit_path,
+                unit_path,
                 _job_id,
                 _job_type,
                 _job_path,
             )| {
-                if following.is_empty() && active_state != "inactive" {
-                    Some((id, active_state, sub_state))
+                let proxy = systemd_manager.connection.with_proxy(
+                    "org.freedesktop.systemd1",
+                    unit_path,
+                    Duration::from_millis(5000),
+                );
+
+                if following.is_empty() && state != "inactive" {
+                    Some((
+                        id,
+                        UnitState {
+                            state,
+                            substate,
+                            proxy,
```

Comment:
I wonder if it's a good idea to store this proxy or if we should just extract the info we want here immediately and store that.

Reply:
I implemented it like that originally, then did a small refactor to just pass around the proxy, as this felt nicer since the proxy can be used to do any dbus-related things on that unit later on. I'm cool with either though.

Reply:
As long as there's no resource penalty then sure. I'm not familiar enough with dbus to know if having a whack ton of objects laying around can be problematic.